klippa/src/lib.rs

```diff
@@ -356,7 +467,19 @@ pub fn estimate_subset_table_size(font: &FontRef, table_tag: Tag, plan: &Plan) -
     let same_size: bool =
         table_tag == Gsub::TAG || table_tag == Gpos::TAG || table_tag == Name::TAG;
 
-    //TODO: Add extra room retain_gids
+    if plan
+        .subset_flags
+        .contains(SubsetFlags::SUBSET_FLAGS_RETAIN_GIDS)
+    {
+        if table_tag == Cff::TAG {
+            //Add some extra room for the CFF charset
+            bulk += src_glyphs * 16;
+        } else if table_tag == Cff2::TAG {
+            // Just extra CharString offsets
+            bulk += src_glyphs * 4;
```

Comment:
I think it's useful to define named constants for these magic numbers, as a form of documentation. What are 16 and 4 the lengths of?

Reply:
unfortunately I'm also not aware of what these numbers are for, I just ported them from HB.:)

Reply:
I think it's worth the investment to figure out the answer and document in the form of constants. We presumably have code that puts something of the indicated size in place somewhere, no?

Reply:
Per f2f, ok to file an issue and land this. @behdad might know off the top of his head.

Reply:
The CharStrings are stored in an INDEX object. While empty CharStrings take no room, their offset is still stored. The offsets each take at most 4 bytes, hence the value here.

For CFF1, we add more, because the CharSet stores glyph names... These are all heuristics to reduce the need for rework with a larger buffer.

Reply:
Thank you!